File: backend/app/schemas/entity_intel.py
"""Schemas for Entity Intelligence endpoints."""
from typing import Optional
from pydantic import BaseModel, Field


class EntityIntelBrief(BaseModel):
    """AI-generated intelligence brief for an entity."""
    entity_id: str
    entity_type: str
    summary: str
    threat_level: str = Field(description="low | medium | high | critical")
    capabilities: list[str] = []
    mission_profile: Optional[str] = None
    command_control: Optional[str] = None
    confidence: float = Field(default=0.5, ge=0.0, le=1.0)


class EntitySpecification(BaseModel):
    """A single specification entry."""
    key: str
    value: str
    unit: Optional[str] = None


class EntityLink(BaseModel):
    """A relationship between entities."""
    related_entity_id: str
    related_entity_name: str
    related_entity_type: str
    relationship: str


class EntityTimelineEntry(BaseModel):
    """A single event in entity's activity timeline."""
    timestamp: str
    event: str
    detail: Optional[str] = None

File: frontend/src/lib/api/entityIntel.ts
import { ApiClient } from '@/lib/api';
import type { EntityIntelBrief, EntitySpecEntry, EntityLink, EntityTimelineEntry } from '@/lib/store/entityIntel';

const client = new ApiClient();

export const entityIntelApi = {
  async getBrief(entityType: string, entityId: string): Promise<EntityIntelBrief> {
    const data = await client.get<{
      summary: string;
      threat_level: string;
      capabilities: string[];
      mission_profile: string | null;
      command_control: string | null;
      confidence: number;
    }>(`/entity-intel/${entityType}/${entityId}/brief`);
    return {
      summary: data.summary,
      threatLevel: data.threat_level as EntityIntelBrief['threatLevel'],
      capabilities: data.capabilities,
      missionProfile: data.mission_profile,
      commandControl: data.command_control,
      confidence: data.confidence,
    };
  },

  async getSpecs(entityType: string, entityId: string): Promise<EntitySpecEntry[]> {
    const data = await client.get<{ key: string; value: string; unit?: string }[]>(
      `/entity-intel/${entityType}/${entityId}/specs`,
    );
    return data;
  },

  async getLinks(entityType: string, entityId: string): Promise<EntityLink[]> {
    const data = await client.get<
      {
        related_entity_id: string;
        related_entity_name: string;
        related_entity_type: string;
        relationship: string;
      }[]
    >(`/entity-intel/${entityType}/${entityId}/links`);
    return data.map((d) => ({
      relatedEntityId: d.related_entity_id,
      relatedEntityName: d.related_entity_name,
      relatedEntityType: d.related_entity_type,
      relationship: d.relationship,
    }));
  },

  async getTimeline(entityType: string, entityId: string): Promise<EntityTimelineEntry[]> {
    return client.get<EntityTimelineEntry[]>(
      `/entity-intel/${entityType}/${entityId}/timeline`,
    );
  },
};
